developers/weaviate/client-libraries/typescript/v2_v3_migration.md

```diff
@@ -400,7 +400,158 @@ response.data?.Get?.Article?.[0]['_additional']?.creationTimeUnix // Get the tim
 
 ## How to migrate your code
 
-For code examples, see the pages here:
+
+This workflow describes the process one would follow to migrate their codebase to the v3 client. 
+
+1. Install the v3 client package. Update the client package from the v2 to v3 client. 
+
+2. Connect to Weaviate. Make sure your client instantiation respects the new clients syntax. Our documentation on instantiation covers use cases from local, custom to WCS.
+
+3. Migrate your code snippets. The v3 client is collection first which means we do not have to duplicate mentions of the collection you are interacting with.
+
+Below is an example of a basic script in both the v2 and v3 clients. In this script, we define a collection, insert then query objects from the collection, and finally delete the collection. This is to give you an idea of what the migration would entail.
+
+<Tabs groupId="languages">
+<TabItem value="jsv3" label="JS/TS (v3)">
+
+```ts
```

Comment:
Would you highlight the areas that change:

```
// highlight-start
// highlight-end
```

Reply:
Great point!

Reply:
everything kind of changes, adding comments to the code will do a better job explaining what sections of the code do what and people can conclude on their own, what changed.

Reply:
no problem